stdlib/public/Concurrency/DiscardingTaskGroup.swift

```diff
@@ -0,0 +1,554 @@
+/// of side-effect.
+///
+/// A group waits for all of its child tasks
+/// to complete before it returns. Even cancelled tasks must run until
+/// completion before this function returns.
+/// Cancelled child tasks cooperatively react to cancellation and attempt
+/// to return as early as possible.
+/// After this function returns, the task group is always empty.
+///
+/// It is not possible to explicitly await completion of child-tasks,
+/// however the group will automatically await *all* child task completions
+/// before returning from this function:
+///
+/// ```
+/// await withDiscardingTaskGroup { group in
+///   group.addTask { /* slow-task */ }
+///   // slow-task executes...
+/// }
+/// // guaranteed that slow-task has completed and the group is empty & destroyed
+/// ```
+///
+/// Task Group Cancellation
+/// =======================
+///
+/// You can cancel a task group and all of its child tasks
+/// by calling the ``TaskGroup/cancelAll()`` method on the task group,
+/// or by canceling the task in which the group is running.
+///
+/// If you call `addTask(priority:operation:)` to create a new task in a canceled group,
+/// that task is immediately canceled after creation.
+/// Alternatively, you can call `asyncUnlessCancelled(priority:operation:)`,
+/// which doesn't create the task if the group has already been canceled
+/// Choosing between these two functions
+/// lets you control how to react to cancellation within a group:
+/// some child tasks need to run regardless of cancellation,
+/// but other tasks are better not even being created
+/// when you know they can't produce useful results.
+///
+/// Because the tasks you add to a group with this method are nonthrowing,
+/// those tasks can't respond to cancellation by throwing `CancellationError`.
+/// The tasks must handle cancellation in some other way,
+/// such as returning the work completed so far, returning an empty result, or returning `nil`.
+/// For tasks that need to handle cancellation by throwing an error,
+/// use the `withThrowingDiscardingTaskGroup(returning:body:)` method instead.
+///
+/// - SeeAlso: ``withThrowingDiscardingTaskGroup(returning:body:)
+@available(SwiftStdlib 5.8, *)
+@inlinable
+@_unsafeInheritExecutor
+public func withDiscardingTaskGroup<GroupResult>(
+  returning returnType: GroupResult.Type = GroupResult.self,
+  body: (inout DiscardingTaskGroup) async -> GroupResult
+) async -> GroupResult {
+  #if compiler(>=5.5) && $BuiltinTaskGroupWithArgument
+  let flags = taskGroupCreateFlags(
+    discardResults: true
+  )
+
+  let _group = Builtin.createTaskGroupWithFlags(flags, GroupResult.self)
+  var group = DiscardingTaskGroup(group: _group)
+  defer { Builtin.destroyTaskGroup(_group) }
+
+  let result = await body(&group)
+
+  try! await group.awaitAllRemainingTasks() // try!-safe, cannot throw since this is a non throwing group
+  return result
+  #else
+  fatalError("Swift compiler is incompatible with this SDK version")
```

Comment:
Since these are compile-time conditions, could this instead be a compile-time error?

`#error("Swift compiler is incompatible...")`

Reply:
Maybe, but that's what we do everywhere else so I don't really want to change this during this PR.